### PR TITLE
Ensure every cflag is not None and string type when printing

### PR DESCRIPTION
**Pull request by iKlotho**
Fixes https://github.com/unbit/uwsgi/issues/2630

### CODE REVIEW

File: uwsgiconfig.py [hunk 1]
```diff
@@ -336,6 +336,7 @@ def build_uwsgi(uc, print_only=False, gcll=None):
     if 'APPEND_CFLAGS' in os.environ:
         cflags += os.environ['APPEND_CFLAGS'].split()
 
+    cflags = [str(flag) for flag in cflags if flag is not None]
```
Comment (Wwarrior1) [inline uwsgiconfig.py:339]: LGTM

**Review by Wwarrior1 — APPROVED** [review]

**Comment by xrmx** [comment]
I think the bug is in the pcre handling and this is just papering over the bug

**Comment by iKlotho** [comment]
I checked the failed build, and the `pcre_cflags` returns None. Checking the value before appending it to the list could fix the problem. 

```py
if pcre_flags:
    self.cflags.append(pcre_cflags)
```

Another fix would be to modify the `append` method of the existing `list` type.  This way we won't need to check every flag if it is `None` but I am not so sure about this approach.

```py
class SafeList(list):
    def append(self, value):
        if value is not None:
            super().append(item)
```

**Comment by xrmx** [comment]
@iKlotho https://github.com/unbit/uwsgi/issues/2630#issuecomment-2055848014

**Comment by iKlotho** [comment]
> @iKlotho see [#2630 (comment)](https://github.com/unbit/uwsgi/issues/2630#issuecomment-2055848014)

Okay, I see the `pcre` is set to `auto` and `pcre_libs` is not defined. I opened another [PR](https://github.com/unbit/uwsgi/pull/2632) but I am not sure if this case should exit the build. I tried to build with the guard and it succeeded.

**Pull request closed without merging**